tests/functional/workloads/cnv/test_cnv_cluster_shutdown_recovery.py

```diff
@@ -0,0 +1,195 @@
+            source_csums[vm_obj.name] = source_csum
+
+        # Choose VMs randomaly
+        vm_obj, vm_for_stop, vm_for_snap = random.sample(all_vms, 3)
+
+        # Create VM using cloned pvc of source VM PVC
+        vm_obj.stop()
+        clone_obj = clone_vm_workload(
+            vm_obj=vm_obj,
+            volume_interface=vm_obj.volume_interface,
+            namespace=(
+                vm_obj.namespace
+                if vm_obj.volume_interface == constants.VM_VOLUME_PVC
+                else None
+            ),
+        )
+        all_vms.append(clone_obj)
+        csum = cal_md5sum_vm(vm_obj=clone_obj, file_path=file_paths[0])
+        source_csums[clone_obj.name] = csum
+
+        # Create a snapshot
+        # Taking Snapshot of PVC
+        pvc_obj = vm_for_snap.get_vm_pvc_obj()
+        snap_obj = snapshot_factory(pvc_obj)
+
+        # Restore the snapshot
+        res_snap_obj = snapshot_restore_factory(
+            snapshot_obj=snap_obj,
+            storageclass=vm_for_snap.sc_name,
+            volume_mode=snap_obj.parent_volume_mode,
+            access_mode=vm_for_snap.pvc_access_mode,
+            status=constants.STATUS_BOUND,
+            timeout=300,
+        )
+
+        # Create new VM using the restored PVC
+        res_vm_obj = cnv_workload(
+            source_url=constants.CNV_FEDORA_SOURCE,
+            storageclass=vm_for_snap.sc_name,
+            existing_pvc_obj=res_snap_obj,
+            namespace=vm_obj.namespace,
+        )
+        all_vms.append(res_vm_obj)
+        csum = cal_md5sum_vm(vm_obj=res_vm_obj, file_path=file_paths[0])
+        source_csums[res_vm_obj.name] = csum
+
+        # Keep vms in different states (power on, paused, stoped)
+        vm_for_stop.stop()
+        vm_for_snap.pause()
+
+        # Initiate abrupt shutdown the cluster nodes as per OCP official documentation
+        worker_nodes = get_nodes(node_type="worker")
+        master_nodes = get_nodes(node_type="master")
+
+        logger.info("Abruptly/Gracefully Shutting down worker & master nodes")
+        nodes.stop_nodes(nodes=worker_nodes, force=force)
+        nodes.stop_nodes(nodes=master_nodes, force=force)
+
+        logger.info("waiting for 5 min before starting nodes")
+        time.sleep(300)
+
+        # Initate ordered start of cluster after 10 min by following OCP official documentation.
+        logger.info("Starting worker & master nodes")
+        nodes.start_nodes(nodes=master_nodes)
+        nodes.start_nodes(nodes=worker_nodes)
+        retry(
+            (
+                CommandFailed,
+                TimeoutError,
+                AssertionError,
+                ResourceWrongStatusException,
+            ),
+            tries=30,
+            delay=15,
+        )(wait_for_nodes_status(timeout=1800))
+        logger.info("All nodes are now in READY state")
+
+        logger.info("Waiting for 10 min for all pods to come in running state.")
+        time.sleep(600)
```

Comment:
can this sleep be avoided?

Reply:
No we have to wait.

Reply:
we did same in one  the system test to check all pods are in running state.